apps/system/js/sync_manager.js

```diff
@@ -0,0 +1,460 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+'use strict';
+
+/* global BaseModule */
+/* global ERROR_GET_FXA_ASSERTION */
+/* global ERROR_REQUEST_SYNC_REGISTRATION */
+/* global ERROR_SYNC_APP_GENERIC */
+/* global ERROR_SYNC_APP_KILLED */
+/* global ERROR_SYNC_REQUEST */
+/* global FxAccountsClient */
+/* global LazyLoader */
+/* global Service */
+/* global SettingsHelper */
+/* global SyncUnrecoverableErrors */
+/* global uuid */
+
+/**
+ * This module manages the lifecycle of Firefox Sync on Firefox OS.
+ * It connects the Settings application where the Firefox Sync UI is presented
+ * to the user with the Sync app that is responsible of doing the actual
+ * synchronization of the user's data.
+ *
+ * SyncManager reacts to changes done on the Settings Sync UI that triggers
+ * IAC requests or Settings API changes and translates these changes into
+ * actions. It makes use of the SyncStateMachine to control that this actions
+ * are possible on the current situation of Sync and communicates with the
+ * Sync application via IAC API messages when needed.
+ */
+
+(function() {
+  const FXA_EVENT = 'mozFxAccountsUnsolChromeEvent';
+
+  const SYNC_TASK = 'gaia::system::firefoxsync';
+  const SYNC_REQUEST_IAC_KEYWORD = 'gaia::sync::request';
+
+  const HISTORY = 'history';
+  const PASSWORDS = 'passwords';
```

Comment:
Maybe make this an array, so we don't have any code that is particular to a specific collection. In fact, it would be nice if the Systems app would be entirely agnostic to collection names (it is possible to enumerate `sync.collection` from settings, and get the collection names from there?